### DISCUSSION TITLE
Is it possible to react to / answer messages using matrix-commander?

### DISCUSSION THREAD

**Discussion by Marv93** [8go/matrix-commander#116] [Q&A] [answered]
My usecase is: I want to send some data into a chat, the bot processes the data and then sends something (that depends on the exact data that was given) back in the same chat.

It seems matrix-commander can only print messages it received and not invoke any actions (calling other scripts with data from the received message for example). Then I saw that one can import matrix-commander into Python scripts which should enable processing received messages and sending something back. But there seems to be no real documentation, only the example `test-send.py` file which wasn't really helpful.

Is there some way one can use matrix-commander to react to messages? Thanks in advance.

**Comment by edwinsage** [comment, marked as answer]
The simple way is to have your own Python program run matrix-commander as a shell command, and wait for input.  When MC prints some text your program can receive it, figure out what it wants to do, then call another instance of MC to send a message of your choice.  This is the main way I make use of MC, though I use Perl myself and am not sure of the details of how to do it in Python.

**Reply by Marv93** [reply]
Ok so there is no native way built into MC, but as long as I can read the stdout of MC I can answer to it.

In the meantime I solved the problem without MC but I will still mark this as answer as this seems to be the intended way to use MC for this usecase.

**Comment by 8go** [comment]
@edwinsage is correct. 

`matrix-commander` has "receive" and "send" but no "if receive this then send that". 

You follow the use-case from @edwinsage or you use a Matrix bot. 

There are several bots out there including https://github.com/8go/matrix-eno-bot.

**Reply by Marv93** [reply]
I solved the problem by writing my own bot as MC didn't seem to be intended for that. But your answer may still be helpful for people who don't want to write their own bots :+1:
